### PR TITLE
implement `putpmsg`

### LINKED ISSUE #66979: posix: implement `putpmsg()`
This is part of the See [POSIX Roadmap for LTSv3](https://github.com/zephyrproject-rtos/zephyr/issues/51211) (RFC #51211).

`putpmsg()` is required as part of `_XOPEN_STREAMS` Option Group.

For more information, please refer to https://pubs.opengroup.org/onlinepubs/9699919799/functions/putpmsg.html

### PR DESCRIPTION
**Pull request by AbhinavMir**
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/66979

### CODE REVIEW

**Comment by AbhinavMir** [comment]
Sorry for the third PR, I kept getting the branch convention wrong and wanted to keep it clean!

**Comment by ycsin** [comment]
> wanted to keep it clean!

I wouldn't stress about a dirty branch too much, just make sure that the commit messages contain all the necessary purpose for archaeology purposes

File: include/zephyr/posix/stropts.h [hunk 1]
```diff
@@ -19,6 +19,8 @@ struct strbuf {
 
 int putmsg(int fildes, const struct strbuf *ctlptr, const struct strbuf *dataptr, int flags);
```
Comment (ycsin) [inline include/zephyr/posix/stropts.h:21]: the empty line here looks unnecessary

File: lib/posix/CMakeLists.txt [hunk 2]
```diff
@@ -2,3 +2,70 @@
 
 add_subdirectory(options)
 add_subdirectory(shell)
+
```
Comment (ycsin) [inline lib/posix/CMakeLists.txt:5]: looks like a rebase error to me, just create a new branch of the latest `main` (make sure you synced the upstream), and work from there

**Pull request closed without merging**